// evergreen-ui/commonjs/theme/src/default-theme/component-specific/getRowClassName.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _tinycolor = _interopRequireDefault(require("tinycolor2"));

var _themer = require("../../../../themer");

var _memoizeClassName = _interopRequireDefault(require("../utils/memoizeClassName"));

var _scales = _interopRequireDefault(require("../foundational-styles/scales"));

var _palette = _interopRequireDefault(require("../foundational-styles/palette"));

var Appearances = {};
Appearances["default"] = _themer.Themer.createRowAppearance({
  base: {},
  hover: {
    backgroundColor: _scales["default"].neutral.N1A
  },
  focus: {
    backgroundColor: _scales["default"].blue.B1A
  },
  active: {
    backgroundColor: _scales["default"].blue.B2A
  },
  current: {}
});
Appearances.danger = _themer.Themer.createRowAppearance({
  base: {
    backgroundColor: _palette["default"].red.lightest
  },
  hover: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].red.lightest).darken(1).toString()
  },
  focus: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].red.lightest).darken(1.5).toString()
  },
  active: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].red.lightest).darken(2.2).toString()
  },
  current: {}
});
Appearances.warning = _themer.Themer.createRowAppearance({
  base: {
    backgroundColor: _palette["default"].orange.lightest
  },
  hover: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].orange.lightest).darken(1).toString()
  },
  focus: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].orange.lightest).darken(1.5).toString()
  },
  active: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].orange.lightest).darken(2.5).toString()
  },
  current: {}
});
Appearances.success = _themer.Themer.createRowAppearance({
  base: {
    backgroundColor: _palette["default"].green.lightest
  },
  hover: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].green.lightest).darken(1).toString()
  },
  focus: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].green.lightest).darken(2).toString()
  },
  active: {
    backgroundColor: (0, _tinycolor["default"])(_palette["default"].green.lightest).darken(3).toString()
  },
  current: {}
});
/**
 * Get the appearance of a `Row`.
 * @param {string} appearance — only one default appearance.
 * @param {string} intent - none, info, success, warning, danger.
 * @return {string} the appearance object.
 */

var getRowAppearance = function getRowAppearance(appearance, intent) {
  switch (intent) {
    case 'danger':
      return Appearances.danger;

    case 'warning':
      return Appearances.warning;

    case 'success':
      return Appearances.success;

    case 'none':
    default:
      return Appearances["default"];
  }
};
/**
 * Get the className of a `Row`.
 * @param {string} appearance
 * @return {string} the appearance class name.
 */


var _default = (0, _memoizeClassName["default"])(getRowAppearance);

exports["default"] = _default;
